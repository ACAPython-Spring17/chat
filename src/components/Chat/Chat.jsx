import React, { Component } from 'react';
import PropTypes from 'prop-types';

import chat from '../../core/chat';

class Chat extends Component {
  constructor(props) {
    super(props);

    this.offNewMessage = chat.onNewMessage(({ username, message }) => {
      console.log(`Received message from ${username} with message: ${message}`);
    });
  }

  componentWillUnmount() {
    this.offNewMessage();
  }

  render() {
    return (
      <div className="grid-y grid-frame">
        <div className="cell auto cell-block">messages</div>
        <div className="cell shrink">new message input</div>
      </div>
    );
  }
}

Chat.propTypes = {
  username: PropTypes.string.isRequired,
};

export default Chat;
