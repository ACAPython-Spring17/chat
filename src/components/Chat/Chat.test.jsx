import React from 'react';
import Chat from './Chat';

describe('Chat', () => {
  it('should render normally', () => {
    expect(shallow(<Chat username="username" />)).toMatchSnapshot();
  });
});
