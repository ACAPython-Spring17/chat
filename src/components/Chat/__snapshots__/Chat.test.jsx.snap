// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat should render normally 1`] = `
<div
  className="grid-y grid-frame"
>
  <div
    className="cell auto cell-block"
  >
    messages
  </div>
  <div
    className="cell shrink"
  >
    new message input
  </div>
</div>
`;
